snuba/web/rpc/v1/endpoint_trace_item_table.py

```diff
@@ -104,4 +110,12 @@ def _execute(self, in_msg: TraceItemTableRequest) -> TraceItemTableResponse:
                 "This endpoint requires meta.trace_item_type to be set (are you requesting spans? logs?)"
             )
         resolver = self.get_resolver(in_msg.meta.trace_item_type)
-        return resolver.resolve(in_msg)
+        try:
+            return resolver.resolve(in_msg)
+        except QueryException as e:
+            if e.extra[
+                "code"
+            ] == 241 or "DB::Exception: Memory limit (for query) exceeded" in str(e):
+                metrics.increment("endpoint_trace_item_table_OOM")
+                sentry_sdk.capture_exception(e)
+            raise BadSnubaRPCRequestException(str(e))
```

Comment:
I think that's something we'd want for all endpoints, so I'd put that in the base handler.